matching-core/src/main/java/com/graphhopper/matching/util/HmmProbabilities.java

```diff
@@ -67,33 +55,12 @@ public double emissionLogProbability(double distance) {
      *                       consecutive map matching candidates.
      * @param linearDistance Linear distance [m] between two consecutive GPS
      *                       measurements.
-     * @param timeDiff       time difference [s] between two consecutive GPS
-     *                       measurements.
      */
-    public double transitionLogProbability(double routeLength, double linearDistance,
-                                           double timeDiff) {
-        if (timeDiff == 0) {
```

Comment:
>But using the fastest weighting for me implies that we should prefer matching to fast roads even if there are no time stamps attached.

What I tried to explain in the original comment was that the timeDiff==0 check was broken and that the consequences would be even worse for directed candidates so this had to be fixed. The code change above makes sure that penalties from unfavored edges are also considered if all timestamps are equal.

With FastestWeighting and the route length transition metric we are currently using the map matcher favors the shortest routes out of all fastest routes between candidate pairs (independent of whether timestamps are equal or not). This means that the map matcher actually achieves a compromise between fast and short routes. Moreover, we are using the transition metric suggested and verified by [Newson & Krumm](http://www.msr-waypoint.com/en-us/um/people/jckrumm/Publications%202009/map%20matching%20ACM%20GIS%20camera%20ready.pdf). Changing this should be done in a separate PR and carefully evaluated. (This discussion actually belongs into #86.)